### PR TITLE
[Eclipse Foundation] Update to follow branding guidelines

### PR DESCRIPTION
**Pull request by marcdumais-work**
Update repo so that we follow the Eclipse Foundation
branding guidelines. In short:

"the project name needs to be "Eclipse Theia" in the first
and all prominent occurrences, add an indication that "Theia"
is a trademark of the Eclipse Foundation in the README, and
include a link to https://www.eclipse.org/theia."

Signed-off-by: Marc Dumais <marc.dumais@ericsson.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,7 +57,7 @@ Here you can find guides and examples for common scenarios:
 
 ## Contributing
 
-Read below to learn how to take part in improving Theia:
+Read below to learn how to take part in improving Eclipse Theia:
```
Comment (svenefftinge) [inline README.md]: For the sake of simplicity, I'd prefer to leave this as is.

File: README.md [hunk 2]
```diff
@@ -67,7 +68,7 @@ Read below to learn how to take part in improving Theia:
 
 ## Feedback
 
-Read below how to engage with Theia community:
+Read below how to engage with Eclipse Theia community:
```
Comment (svenefftinge) [inline README.md]: For the sake of simplicity, I'd prefer to leave this as is.

File: README.md (hunk 1 above)
Comment (marcdumais-work) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (marcdumais-work) [reply README.md]: done

File: README.md [hunk 3]
```diff
@@ -25,13 +25,14 @@
...
 </div>
 
-- [**Website**](#website)
-- [**Scope**](#scope)
-- [**Getting Started**](#getting-started)
-- [**Contributing**](#contributing)
-- [**Feedback**](#feedback)
-- [**Roadmap**](#roadmap)
-- [**License**](#license)
+- [Website](#website)
```
Comment (marcdumais-work) [inline README.md]: Looks like the VS Code MD extension reformatted this automatically. Will roll-back

**Comment by marcdumais-work** [comment]
@svenefftinge I have done your suggestions, thanks.

**Pull request merged**